pkg/kubernetes/kubernetes.go

```diff
@@ -80,130 +76,162 @@ const (
 	backoffInterval   = 5 * time.Second
 	backoffMaxRetries = 5
 
-	requestTimeout  = 5 * time.Second
-	maxIdleConns    = 1
-	idleConnTimeout = 10 * time.Second
+	defaultQPSLimit   = 100
+	defaultBurstLimit = 150
 )
 
-// ErrEmptyVersionTag Got an empty version tag from GitHub API.
-var ErrEmptyVersionTag = errors.New("got an empty version tag from Github")
+var once sync.Once
 
 // Kubernetes is a client for Kubernetes.
 type Kubernetes struct {
-	client     client.KubeClientConnector
+	k8sClient  ctrlclient.Client
 	l          *zap.SugaredLogger
-	namespace  string
-	httpClient *http.Client
+	rcLock     *sync.Mutex
```

Comment:
I'm not sure we need this mutex. I know we had it previously in the functions for the OLM's InstallPlans and Subscriptions but you removed it from all functions except the `GetInstallPlan` so maybe we should just get rid of it.